framework/src/timeintegrators/NewmarkBeta.C

```diff
@@ -0,0 +1,80 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NewmarkBeta.h"
+#include "NonlinearSystem.h"
+
+registerMooseObject("MooseApp", NewmarkBeta);
+
+template <>
+InputParameters
+validParams<NewmarkBeta>()
+{
+  InputParameters params = validParams<TimeIntegrator>();
+  params.addClassDescription(
+      "Computes the first and second time derivative of variable using Newmark-Beta method.");
+  params.addParam<Real>("beta", 0.25, "beta value");
+  params.addParam<Real>("gamma", 0.5, "gamma value");
+  return params;
+}
+
+NewmarkBeta::NewmarkBeta(const InputParameters & parameters)
+  : TimeIntegrator(parameters),
+    _beta(getParam<Real>("beta")),
+    _gamma(getParam<Real>("gamma")),
+    _u_dot_old(*_sys.solutionUDotOld()),
+    _u_dotdot(_sys.solutionUDotDot()),
+    _u_dotdot_old(*_sys.solutionUDotDotOld()),
+    _du_dotdot_du(_sys.duDotDotDu())
+{
+}
+
+NewmarkBeta::~NewmarkBeta() {}
+
+void
+NewmarkBeta::computeTimeDerivatives()
+{
+  // compute second derivative
+  // according to Newmark-Beta method
+  // u_dotdot = first_term - second_term - third_term
+  //       first_term = (u - u_old) / beta / dt ^ 2
+  //      second_term = u_dot_old / beta / dt
+  //       third_term = u_dotdot_old * (1 / 2 / beta - 1)
+  _u_dotdot = *_solution;
+  _u_dotdot -= _solution_old;
+  _u_dotdot *= 1.0 / _beta / _dt / _dt;
+  _u_dotdot.add(-1.0 / _beta / _dt, _u_dot_old);
+  _u_dotdot.add(-0.5 / _beta + 1.0, _u_dotdot_old);
+
+  // compute first derivative
+  // according to Newmark-Beta method
+  // u_dot = first_term + second_term + third_term
+  //       first_term = u_dot_old
+  //      second_term = u_dotdot_old * (1 - gamma) * dt
+  //       third_term = u_dotdot * gamma * dt
+  _u_dot = _u_dot_old;
+  _u_dot.add((1.0 - _gamma) * _dt, _u_dotdot_old);
+  _u_dot.add(_gamma * _dt, _u_dotdot);
```

Comment:
While this looks like it is implementing (23), which is in terms of the "full" acceleration vectors (i.e. not the increment). Is my understanding correct and if so, why did you mix the two types of updates?

Reply:
Eq. 25 is obtained from Eq. 23 by replacing \ddot{x}_{i+1} by the one calculated in Eqn. 24, and setting \delta \dot{x}_{i} = \dot{x}_{i+1} - \dot{x}_{i}.
 
Since, I already calculated \ddot{x}_{i+1} in lines 50-53, I am just using that value instead of using Eqn. 24. I am calculating the full acceleration and velocity and not the incremental forms.

Reply:
OK, I figured it must be some explanation like that, just wanted to make sure.